test/jdk/java/awt/Checkbox/CheckboxIndicatorSizeTest.java

```diff
@@ -0,0 +1,171 @@
+        frame = new Frame("CheckboxIndicatorSizeTest");
+
+        testPanel = new Panel(new GridLayout(0, 1));
+        testPanel.setFont(new Font("Helvetica", Font.PLAIN, 12));
+        frame.add(testPanel);
+
+        MenuBar menuBar = new MenuBar();
+        Menu fontSizeMenu = new Menu("FontSize");
+
+        MenuItem size10 = new MenuItem("10");
+        size10.addActionListener(this);
+        fontSizeMenu.add(size10);
+
+        MenuItem size12 = new MenuItem("12");
+        size12.addActionListener(this);
+        fontSizeMenu.add(size12);
+
+        MenuItem size14 = new MenuItem("14");
+        size14.addActionListener(this);
+        fontSizeMenu.add(size14);
+
+        MenuItem size18 = new MenuItem("18");
+        size18.addActionListener(this);
+        fontSizeMenu.add(size18);
+
+        MenuItem size24 = new MenuItem("24");
+        size24.addActionListener(this);
+        fontSizeMenu.add(size24);
+
+        MenuItem size36 = new MenuItem("36");
+        size36.addActionListener(this);
+        fontSizeMenu.add(size36);
+
+        menuBar.add(fontSizeMenu);
+        frame.setMenuBar(menuBar);
+
+        Checkbox cbEnglishOnly
+            = new Checkbox("Toggle", true);
+        Checkbox cbDingbatsOnly
+            = new Checkbox("\u274a\u274b\u274c\u274d", true);
+        Checkbox cbEnglishDingbats
+            = new Checkbox("Toggle \u274a\u274d", true);
+        Checkbox cbDingbatsEnglish
+            = new Checkbox("\u274a\u274d toggle", true);
+
+        CheckboxGroup radioGroup = new CheckboxGroup();
+        Checkbox rbEnglishOnly
+            = new Checkbox("Radio", true, radioGroup);
+        Checkbox rbDingbatsOnly
+            = new Checkbox("\u274a\u274b\u274c\u274d", false, radioGroup);
+        Checkbox rbEnglishDingbats
+            = new Checkbox("Radio \u274a\u274d", false, radioGroup);
+        Checkbox rbDingbatsEnglish
+            = new Checkbox("\u274a\u274d radio", false, radioGroup);
+
+        Label cbLabel = new Label("Checkboxes");
+        cbLabel.setBackground(Color.YELLOW);
+        testPanel.add(cbLabel);
+        testPanel.add(cbEnglishOnly);
+        testPanel.add(cbDingbatsOnly);
+        testPanel.add(cbEnglishDingbats);
+        testPanel.add(cbDingbatsEnglish);
+
+        Label rbLabel = new Label("Radio buttons");
+        rbLabel.setBackground(Color.YELLOW);
+        testPanel.add(rbLabel);
+        testPanel.add(rbEnglishOnly);
+        testPanel.add(rbDingbatsOnly);
+        testPanel.add(rbEnglishDingbats);
+        testPanel.add(rbDingbatsEnglish);
+
+        frame.pack();
+        return frame;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        String sizeStr = e.getActionCommand();
+        int size = Integer.parseInt(sizeStr);
+        Font oldFont = testPanel.getFont();
+        Font newFont = new Font(oldFont.getName(), oldFont.getStyle(), size);
+        testPanel.setFont(newFont);
+        frame.invalidate();
+        frame.validate();
```

Comment:
DO we need to invalidate and then validate? guess we can remove these 2 lines or does it have any impact on testing?